garage/tf/models/gaussian_mlp_model2.py

```diff
@@ -0,0 +1,178 @@
+                mean_std_network = mlp(
+                    state_input,
+                    output_dim=action_dim * 2,
+                    hidden_sizes=self._hidden_sizes,
+                    hidden_nonlinearity=hidden_nonlinearity,
+                    output_nonlinearity=output_nonlinearity,
+                    output_b_init=b,
+                    name='mean_std_network')
+                with tf.variable_scope('mean_network'):
+                    mean_network = mean_std_network[..., :action_dim]
+                with tf.variable_scope('std_network'):
+                    std_network = mean_std_network[..., action_dim:]
+
+            else:
+                # separate MLPs for mean and std networks
+                # mean network
+                mean_network = mlp(
+                    state_input,
+                    output_dim=action_dim,
+                    hidden_sizes=self._hidden_sizes,
+                    hidden_nonlinearity=hidden_nonlinearity,
+                    output_nonlinearity=output_nonlinearity,
+                    name='mean_network')
+
+                # std network
+                if self._adaptive_std:
+                    b = tf.constant_initializer(self._init_std_param)
+                    std_network = mlp(
+                        state_input,
+                        output_dim=action_dim,
+                        hidden_sizes=self._std_hidden_sizes,
+                        hidden_nonlinearity=std_hidden_nonlinearity,
+                        output_nonlinearity=std_output_nonlinearity,
+                        output_b_init=b,
+                        name='std_network')
+                else:
+                    p = tf.constant_initializer(self._init_std_param)
+                    std_network = parameter(
+                        state_input,
+                        length=action_dim,
+                        initializer=p,
+                        trainable=self._learn_std,
+                        name='std_network')
+
+        mean_var = mean_network
+        log_std_var = std_network
+
+        with tf.variable_scope('std_parameterization'):
+            # build std_var with std parameterization
+            if self._std_parameterization == 'exp':
+                pass
+            elif self._std_parameterization == 'softplus':
+                softplus_std_var = tf.log(1. + tf.exp(log_std_var))
+                log_std_var = tf.log(softplus_std_var)
+            else:
+                raise NotImplementedError
+
+        with tf.variable_scope('std_limits'):
+            if self._min_std_param:
+                log_std_var = tf.maximum(log_std_var, self._min_std_param)
+            if self._max_std_param:
+                log_std_var = tf.minimum(log_std_var, self._max_std_param)
+
+        distribution = dist(mean_var, tf.exp(log_std_var))
+
+        action_var = distribution.sample(seed=ext.get_seed())
+
+        return action_var, log_std_var, distribution
```

Comment:
why not return `mean_var`?

Reply:
we can get `mean_var` by `distribution.mean()`.

Reply:
this will be easier to make right once we get rid of garage.distributions